### PR TITLE
Feature/logging module

### PR DESCRIPTION
**Pull request by GH-Syn**
## About this merge
 - Adding logging module. 
 - Logs to file 
  - I haven't tested every single log statement but most should work. (will be tested once unit tests are done)
  
 ## Tasks needed before merge
 - [x] Add relevant files / folder to gitignore
 - [x] unit tests for any logging setup files

### CODE REVIEW

File: Config/config.py [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+    """
+    Get logging configuration from configuration file.
+
+    :return: logging configuration as a dictionary
+    :rtype: dict
+    """
+    try:
+        open('Config/logger.ini')
+    except Exception:
+        print("File location invalid  ")
```
Comment (GH-Syn) [inline Config/config.py:41]: - [ ] needs logger call
- [ ] print file location

**Comment by GH-Syn** [comment]
Pending review @initd1

File: Config/formatter.py
Comment (initd1) [inline Config/formatter.py:1]: nice touch with the emoji's

Comment (GH-Syn) [reply Config/formatter.py:1]: haha, thanks :)

**Review by GH-Syn — COMMENTED** [review]
marking review as done.

File: Config/check_config_validity.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import configparser
+
+config = configparser.ConfigParser()
+config.read('Config/logger.ini')
+
+# check if the required sections are present
+if 'loggers' not in config.sections() or 'handlers' not in config.sections():
+    raise ValueError('Missing required section in config.ini')
```
Comment (initd1) [inline Config/check_config_validity.py:8]: I believe this message would have to be:
```python
raise ValueError('Missing required section in logger.ini')
```

File: Config/check_config_validity.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+if 'keys' not in config['loggers'] or 'suspicious' not in config['loggers']:
+    raise ValueError('Missing required key in loggers section in config.ini')
```
Comment (initd1) [inline Config/check_config_validity.py:11]: similar change required:
```python
raise ValueError('Missing required key in loggers section in **logger**.ini')
```

File: Config/check_config_validity.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# check if the required keys are present in the 'handlers' section
+if 'keys' not in config['handlers'] or 'console' not in config['handlers']['keys']:
+    raise ValueError('Missing required key in handlers section in config.ini')
```
Comment (initd1) [inline Config/check_config_validity.py:15]: similar change required:
```python
raise ValueError('Missing required key in handlers section in **logger**.ini')
```

File: Config/check_config_validity.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+# check if the values of the keys are in the expected format
+if not isinstance(config.getint('handlers', 'console.level'), int):
+    raise ValueError('Invalid value for console.level in config.ini')
```
Comment (initd1) [inline Config/check_config_validity.py:19]: similar change:
```python
raise ValueError('Invalid value for console.level in logger.ini')
```

File: Config/check_config_validity.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+if not isinstance(config.getint('loggers', 'keys.suspicious.level'), int):
+    raise ValueError('Invalid value for keys.suspicious.level in config.ini')
```
Comment (initd1) [inline Config/check_config_validity.py:22]: similar change:
```python
raise ValueError('Invalid value for keys.suspicious.level in logger.ini')
```

**Review by initd1 — COMMENTED** [review]
I have encountered some issues while executing these changes in my environment. Apart from that there are some minor issues and typos.

I believe this may have preemptively been merged to main. Apologies if I submitted the review. I was thinking you wouldn't be able to merge to main since I have it as a protected branch.

File: Tests/Config/test_config.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+from configparser import ConfigParser
+
+class TestConfig(unittest.TestCase):
+    def setUp(self):
+        self.path = os.path.join('Config/logger.ini')
+
+        if not os.path.isdir('Logs'):
+            os.mkdir('Logs')
+        if not os.path.exists(os.path.join('Logs', 'traceback.log')):
+            with open(os.path.join('Logs', 'traceback.log')) as fp:
```
Comment (initd1) [inline Tests/Config/test_config.py:12]: I have encountered an issue while running the test when traceback.log file isn't already created. Like in line 10, where Logs directory is created if not exists, we may need to create traceback.log if it doesn't exist.

**Comment by initd1** [comment]
@GH-Syn The other issue is, when the script is run with the correct API Keys,  logs to traceback.log are not being written, and no output is displayed on the terminal.  However, no errors during execution. 

Might have something to do with the file handler and stream handler.

Example:
```python
$ python main.py -e k@k.cm

$
``` 

Don't think a revert of the merge would be recommended, but probably a separate bug issue to resolve these? Let me know your thoughts.

**Comment by initd1** [comment]
> @GH-Syn The other issue is, when the script is run with the correct API Keys, logs to traceback.log are not being written, and no output is displayed on the terminal. However, no errors during execution.
> 
> Might have something to do with the file handler and stream handler.
> 
> Example:
> 
> ```python
> $ python main.py -e k@k.cm
> 
> $
> ```
> 
> Don't think a revert of the merge would be recommended, but probably a separate bug issue to resolve these? Let me know your thoughts.

Update: I have resolved the issue of stream handler  not displaying any output in #31. Will rename the issue and push the update

**Pull request merged**